packages/transformers/babel/src/flow.js

```diff
@@ -1,19 +1,19 @@
 // @flow
-import type {MutableAsset} from '@parcel/types';
+
+import {IConfig} from '@parcel/types';
+import type {BabelConfig} from './types';
 
 /**
  * Generates a babel config for stripping away Flow types.
  */
-export default async function getFlowConfig(asset: MutableAsset) {
-  if (/^(\/{2}|\/\*+) *@flow/.test((await asset.getCode()).substring(0, 20))) {
```

Comment:
Hmm okay, that's something I can look into. What plugin do you mean by the "it still has to add the parser plugin"?

Reply:
The flow plugin actually adds two plugins: the parser plugin to enable parsing flow syntax, and the babel plugin to strip the types off. See https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-flow-strip-types/src/index.js#L14. Last time I tried, enabling the flow syntax parsing was quite a bit slower than not so keeping it off when not needed might be good for performance. Also, I believe the flow parser might introduce some ambiguities with how JS is parsed, so it would be more spec compliant to keep it off unless it is enabled with the comment.

Reply:
Okay yeah, probably still worth it to do then. Problem right now is that we don't have access to asset in the main process right now. I mentioned in another [comment](https://github.com/parcel-bundler/parcel/pull/3261#discussion_r310237543) that I'm thinking of ways to solve for that. Is it cool if we leave this problem for a followup?